File: Sem-5/DAA/BinarySearch.c
// Binary Search in C
#include <stdio.h>
void main(){
    int arr[] = {2, 3, 5, 6, 8};
    int n = sizeof(arr) / sizeof(arr[0]);
    int key = 6;
    int left = 0, right = n - 1;
    int found = 0;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == key) {
            printf("Element %d found at index %d\n", key, mid);
            found = 1;
            break;
        } else if (arr[mid] < key) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    if (!found) {
        printf("Element %d not found in the array\n", key);
    }
}

File: Sem-3/Data Structure/SearchAndShort/BinarySearch.java

import java.util.Scanner;

public class BinarySearch {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr[] = new int[n];
        for(int i=0 ; i<n ; i++){
            arr[i] = sc.nextInt();
            
        }

        System.out.println("Which element would you like to find");
        int a = sc.nextInt();
        int first = 0;
        int last = arr.length-1;
        int mid = 0 ;

        while(last>=first){
            if(arr[first]==a || arr[last]==a){
                System.out.println("Number found in array ");
                return;
            }

            mid = (first+last)/2;
            if(arr[mid] == a){
                System.out.println("Number found in the array at index : "+mid);
                return;
            }
            else if(arr[mid] > a){
                last = mid-1;
            }
            else if(arr[mid] < a){
                first = mid+1 ;
            }
        }
        System.out.println("Number not found in the array");
    }    
}
